### ISSUE TITLE
`solid/reactivity` hardcodes the assumption that solid primitives are imported from `"solid-js"`

### ISSUE THREAD

**Issue by Alloyed** [solidjs-community/eslint-plugin-solid#183]
**Describe the need**
semi-related to: https://github.com/solidjs/solid-refresh/issues/44
In our codebase, we use a universal renderer that explicitly skips the web-specific features present in `"solid-js"`. this means, instead of writing `import {createSignal} from 'solid-js'`, we write `import {createSignal} from 'my-custom-solid-renderer';`

This kind of accidentally worked in earlier versions of eslint-plugin-solid, but it seems that in-between 0.13.2 and 0.14.5, the symbol detection got more strict, leading to many more false negatives.

**Suggested Solution**
some kind of config option to specify the import path, eg. solidImportPattern: /^my-custom-solid-renderer/
this would involve adding a parameter to the `trackImports()` method and adding this config flag to every rule that uses it. (I don't know eslint that well, but if this could be a global parser flag ala the tsconfig that would probably be better)

https://github.com/solidjs-community/eslint-plugin-solid/blob/main/packages/eslint-plugin-solid/src/utils.ts#L149

**Possible Alternatives**
the upstream solid package could be refactored to not be web-specific, or for the reactive pieces to exist in their own library (https://github.com/solidjs/solid/issues/2051). i think this is already happening for solid 2.

we could remove the requirement altogether that imports come from the solid-js library, but i think that would be worse for all other users.

- [x] I would be willing to contribute a PR to implement this feature

**Comment by ryansolid** [comment]
Accepted direction: 0.16 introduced `settings.solid` (currently `version`), and custom import sources fit naturally there — e.g. `settings: { solid: { imports: ["my-custom-solid-renderer"] } }` feeding `trackImports` across all rules, rather than per-rule options. Universal renderers are a first-party concern for Solid 2.0, so this is on the roadmap.

**Comment by ryansolid** [comment]
Shipped in 0.18.0: add your module names to `settings.solid.moduleSources` and the import-gated rules will treat their exports as Solid's primitives:

```js
{
  settings: { solid: { moduleSources: ["my-custom-solid-renderer"] } },
}
```

Names are matched whole against the import source. This applies to all rules that resolve imports through the shared tracker (reactivity, no-unused-signal, no-write-in-pure-computation, no-store-mutation-outside-setter, no-single-arg-create-effect, and friends). Note the `reactivity` rule was already permissive about `create*`/`use*` naming regardless of import source, so the setting mostly benefits the stricter 2.0-era rules.
